scylla/src/transport/topology.rs

```diff
@@ -749,3 +818,94 @@ fn strategy_from_string_map(
+        let test_cases = [
+            ("bigint", CqlType::Native(NativeType::BigInt)),
+            (
+                "list<int>",
+                CqlType::Collection {
+                    frozen: false,
+                    type_: CollectionType::List(Box::new(CqlType::Native(NativeType::Int))),
+                },
+            ),
+            (
+                "set<ascii>",
+                CqlType::Collection {
+                    frozen: false,
+                    type_: CollectionType::Set(Box::new(CqlType::Native(NativeType::Ascii))),
+                },
+            ),
+            (
+                "map<blob, boolean>",
+                CqlType::Collection {
+                    frozen: false,
+                    type_: CollectionType::Map(
+                        Box::new(CqlType::Native(NativeType::Blob)),
+                        Box::new(CqlType::Native(NativeType::Boolean)),
+                    ),
+                },
+            ),
+            (
+                "frozen<map<text, text>>",
+                CqlType::Collection {
+                    frozen: true,
+                    type_: CollectionType::Map(
+                        Box::new(CqlType::Native(NativeType::Text)),
+                        Box::new(CqlType::Native(NativeType::Text)),
+                    ),
+                },
+            ),
+            (
+                "tuple<tinyint, smallint, int, bigint, varint>",
+                CqlType::Tuple(vec![
+                    CqlType::Native(NativeType::TinyInt),
+                    CqlType::Native(NativeType::SmallInt),
+                    CqlType::Native(NativeType::Int),
+                    CqlType::Native(NativeType::BigInt),
+                    CqlType::Native(NativeType::Varint),
+                ]),
+            ),
+            (
+                "com.scylladb.types.AwesomeType",
+                CqlType::UserDefinedType {
+                    frozen: false,
+                    name: "com.scylladb.types.AwesomeType".to_string(),
+                },
+            ),
+            (
+                "frozen<ks.my_udt>",
+                CqlType::UserDefinedType {
+                    frozen: true,
+                    name: "ks.my_udt".to_string(),
+                },
+            ),
+            (
+                "map<text, frozen<map<text, text>>>",
+                CqlType::Collection {
+                    frozen: false,
+                    type_: CollectionType::Map(
+                        Box::new(CqlType::Native(NativeType::Text)),
+                        Box::new(CqlType::Collection {
+                            frozen: true,
+                            type_: CollectionType::Map(
+                                Box::new(CqlType::Native(NativeType::Text)),
+                                Box::new(CqlType::Native(NativeType::Text)),
+                            ),
+                        }),
+                    ),
+                },
+            ),
+        ];
```

Comment:
Just to verify that the parser doesn't get stuck on deeper types, let's add one use case for an abomination like `map<frozen<list>, set<list<tuple<list<list<....`

Reply:
Phew, done